sale_order_product_recommendation_packaging_default/wizards/sale_order_recommendation.py

```diff
@@ -109,3 +111,22 @@ def _prepare_new_so_line_vals(self, sequence):
             # Nothing to create
             return result
         return self._prepare_packaging_line_vals(result)
+
+    @api.depends("product_id")
+    def _compute_require_packaging(self):
+        for line in self:
+            line.require_packaging = line.product_id.require_packaging
+
+    @api.depends("product_id", "product_uom_qty", "product_uom")
+    def _warning_product_packaging_required(self):
+        """Notify the user when a packaging is required."""
+        result = {}
+        for line in self:
+            if line.require_packaging and not line.product_packaging_id:
+                raise UserError(
```

Comment:
issue: [within constrains, you should raise `ValidationError`s](https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html?highlight=constrains#odoo.api.constrains):

```suggestion
                raise ValidationError(
```